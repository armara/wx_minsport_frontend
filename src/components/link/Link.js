import React from 'react'
import PropTypes from 'prop-types'

import 'components/link/Link.scss'

const Link = ({ href, children, className, target, onClick }) => (
  <a
    href={href}
    className={className}
    target={target}
    rel="noreferrer noopener"
    onClick={onClick}
  >
    {children}
  </a>
)

Link.defaultProps = {
  children: '',
  className: '',
  href: '#',
  target: '_blank',
  onClick: () => {},
}

Link.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  target: PropTypes.oneOf(['_blank', '_self', '_parent', '_top']),
  onClick: PropTypes.func,
  href: PropTypes.string,
  className: PropTypes.string,
}

export default Link
