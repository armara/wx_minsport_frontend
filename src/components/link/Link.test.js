import React from 'react'
import Link from 'components/link/Link'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<Link />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<Link href="test" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly mock props', () => {
    const tree = renderer
      .create(
        <Link
          href="/"
          target="_self"
          onClick={() => {}}
          className="embassy_konigsberg"
        >
          <span>Peters Grand Embassy</span>
        </Link>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
