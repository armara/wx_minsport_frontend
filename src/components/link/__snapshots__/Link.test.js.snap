// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly mock props 1`] = `
<a
  className="embassy_konigsberg"
  href="/"
  onClick={[Function]}
  rel="noreferrer noopener"
  target="_self"
>
  <span>
    Peters Grand Embassy
  </span>
</a>
`;

exports[`<Link /> renders correctly with default props 1`] = `
<a
  className=""
  href="test"
  onClick={[Function]}
  rel="noreferrer noopener"
  target="_blank"
>
  
</a>
`;
